src/v/cluster/security_frontend.cc

```diff
@@ -339,4 +339,98 @@ security_frontend::get_bootstrap_user_creds_from_env() {
       std::in_place, std::move(username), std::move(credentials));
 }
 
+std::vector<acl_resource> security_frontend::do_describe_acls(
+  const security::acl_binding_filter& filter) {
+    /*
+     * collapse common acls by pattern
+     */
+    absl::flat_hash_map<
+      security::resource_pattern,
+      std::vector<security::acl_entry>>
+      entries;
+
+    auto bindings = _authorizer.local().acls(filter);
+
+    for (const auto& binding : bindings) {
+        entries[binding.pattern()].push_back(binding.entry());
+    }
+
+    std::vector<acl_resource> resources;
+    resources.reserve(entries.size());
+    for (const auto& entry : entries) {
+        acl_resource resource;
+
+        // resource pattern
+        resource.type = entry.first.resource();
+        resource.name = entry.first.name();
+        resource.pattern_type = entry.first.pattern();
+
+        // acl entries
+        for (auto& acl : entry.second) {
+            // ignore ephemeral_users
+            auto ephemeral_user = security::principal_type::ephemeral_user;
+            if (acl.principal().type() == ephemeral_user) {
+                continue;
+            }
+            acl_description desc{
+              .principal = acl.principal(),
+              .host = acl.host(),
+              .operation = acl.operation(),
+              .permission_type = acl.permission(),
+            };
+            resource.acls.push_back(std::move(desc));
+        }
+        resources.push_back(std::move(resource));
```

Comment:
The `emplace_back` suggestion was marked resolved without comment, so just wondered if it was forgotten.